scripts/jest/installGatedTestHelpers.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+/* eslint-disable no-for-of-loops/no-for-of-loops */
+
+const noop = () => {};
+
+async function expectTestToFail(callback, errorMsg) {
+  if (callback.length > 0) {
+    throw Error(
+      'Gated test helpers do not support the `done` callback. Return a ' +
+        'promise instead.'
+    );
+  }
+  try {
+    const maybePromise = callback();
+    if (
+      maybePromise !== undefined &&
+      maybePromise !== null &&
+      typeof maybePromise.then === 'function'
+    ) {
+      await maybePromise;
+    }
+  } catch (error) {
+    // Failed as expected
+    return;
+  }
+  throw Error(errorMsg);
+}
+
+function readFlag(flags, flagName) {
+  if (flagName.startsWith('!')) {
+    flagName = flagName.slice(1);
+    const flagValue = flags[flagName];
+    if (flagValue === undefined) {
+      throw Error('Missing feature flag: ' + flagName);
+    }
+    if (flagValue === true) {
+      return `Test is expected to fail because ${flagName} is true.`;
+    }
+  } else {
+    const flagValue = flags[flagName];
+    if (flagValue === undefined) {
+      throw Error('Missing feature flag: ' + flagName);
+    }
+    if (flagValue === false) {
+      return `Test is expected to fail because ${flagName} is false.`;
+    }
+  }
+  return null;
+}
+
+function testGate(testFn, flagNames, testName, fn) {
+  const featureFlags = require('shared/ReactFeatureFlags');
+
+  let errorMessage;
+  if (typeof flagNames === 'string') {
+    errorMessage = readFlag(featureFlags, flagNames);
+  } else if (Array.isArray(flagNames)) {
+    for (const flagName of flagNames) {
+      errorMessage = readFlag(featureFlags, flagName);
+      if (errorMessage !== null) {
+        break;
+      }
+    }
+  } else if (typeof flagNames === 'function') {
+    const shouldPass = flagNames(featureFlags);
+    errorMessage = shouldPass
+      ? null
+      : 'Gated test was expected to fail, but it passed.';
+  } else {
+    throw Error('testGate: Must pass one or more feature flags.');
+  }
+
+  // Expect test to pass
+  if (errorMessage !== null) {
+    testFn(`[GATED, SHOULD FAIL] ${testName}`, () =>
+      expectTestToFail(fn, errorMessage)
+    );
+  } else {
+    testFn(testName, fn);
+  }
+}
+
+function installGatedTestHelpers(channel) {
+  const it = global.it;
+  const fit = global.fit;
+  const xit = global.xit;
```

Comment:
I kind of was hoping to switch to the `test()` form. Does this patch that too? I guess it’s just an alias for `it`?

Reply:
Yeah `it` is an alias